hydromt/models/_v1/model_root.py

```diff
@@ -0,0 +1,187 @@
+"""the new model root class."""
+
+from glob import glob
+from logging import FileHandler, Logger, getLogger
+from os import PathLike, getcwd, getlogin, makedirs
+from os.path import abspath, dirname, isdir, join
+from pathlib import Path, PurePosixPath, PureWindowsPath
+from platform import system
+from typing import List, Optional, Tuple, cast
+
+from hydromt.log import add_filehandler
+from hydromt.typing import ModeLike, ModelMode
+
+logger = getLogger(__name__)
+
+
+class ModelRoot:
+    """A class to handle model roots in a cross platform manner."""
+
+    def __init__(
+        self,
+        path: PathLike,
+        mode: ModeLike = "w",
+        folders: Optional[List[str]] = None,
+        create_dirs: bool = False,
+        logger: Logger = logger,
+    ):
+        self.logger: Logger = logger
+        self.folders = folders or ["."]
+        self.set_root(path, mode, create_dirs)
+
+    def set_root(self, path: PathLike, mode: ModeLike = "w", create_dirs: bool = False):
+        """Initialize the model root.
+
+        In read/append mode a check is done if the root exists.
+        In write mode the required model folder structure is created.
+
+        Parameters
+        ----------
+        root : str, optional
+            path to model root
+        mode : {"r", "r+", "w"}, optional
+            read/append/write mode for model files
+        """
+        self.set_mode(mode)
+        # set path is to take care of cross platform paths
+        self._set_path(path)
+        if self.is_writing_mode():
+            if create_dirs:
+                self._setup_folder_structure()
+                self._setup_log_file_hanglers()
+        if self.is_reading_mode():
+            self._check_root_exists()
+
+    def _check_root_exists(self):
+        # check directory
+        if not isdir(self._path):
+            raise IOError(f'model root not found at "{self._path}"')
+
+    def _setup_folder_structure(self):
```

Comment:
good point, I'll put that in a seperate function

Reply:
Now that we got rid of the folders attribute here it cannot check this anymore, it will have to be done by the model components. We do however, check that the model root itself exists